src/ReaderThemeSupportFeatures.php

```diff
@@ -203,13 +256,35 @@ public function update_cached_theme_support() {
 	 */
 	public function get_theme_support_features( $reduced = false ) {
 		$features = [];
+
 		foreach ( array_keys( self::SUPPORTED_FEATURES ) as $feature_key ) {
-			$feature_value = current( (array) get_theme_support( $feature_key ) );
+			if ( $this->theme_has_theme_json() ) {
+				$feature_value   = [];
+				$global_settings = wp_get_global_settings( self::SUPPORTED_THEME_JSON_FEATURES[ $feature_key ], 'theme' );
+
+				if ( isset( $global_settings[ self::KEY_THEME ] ) ) {
+					$feature_value = array_merge( $feature_value, $global_settings[ self::KEY_THEME ] );
+				}
+
+				if ( isset( $global_settings[ self::KEY_DEFAULT ] ) ) {
+					$feature_value = array_merge( $feature_value, $global_settings[ self::KEY_DEFAULT ] );
+				}
```

Comment:
I confess I am not up to speed on `theme.json`, but wouldn't the `theme` value supercede the `default` value? In other words, shouldn't these two conditions be swapped? (I'd be grateful to be educated otherwise!)

Reply:
The following is the output (in array) when we use `wp_get_global_settings` to retrieve theme features from `theme.json`:

Here, `theme` is derived from `theme.json`, and the `default` preset is coming from the block editor.

<details><summary>Theme Features</summary>

```json
{
  "default": [
    {
      "name": "Black",
      "slug": "black",
      "color": "#000000"
    },
    {
      "name": "Cyan bluish gray",
      "slug": "cyan-bluish-gray",
      "color": "#abb8c3"
    },
    {
      "name": "White",
      "slug": "white",
      "color": "#ffffff"
    },
    {
      "name": "Pale pink",
      "slug": "pale-pink",
      "color": "#f78da7"
    },
    {
      "name": "Vivid red",
      "slug": "vivid-red",
      "color": "#cf2e2e"
    },
    {
      "name": "Luminous vivid orange",
      "slug": "luminous-vivid-orange",
      "color": "#ff6900"
    },
    {
      "name": "Luminous vivid amber",
      "slug": "luminous-vivid-amber",
      "color": "#fcb900"
    },
    {
      "name": "Light green cyan",
      "slug": "light-green-cyan",
      "color": "#7bdcb5"
    },
    {
      "name": "Vivid green cyan",
      "slug": "vivid-green-cyan",
      "color": "#00d084"
    },
    {
      "name": "Pale cyan blue",
      "slug": "pale-cyan-blue",
      "color": "#8ed1fc"
    },
    {
      "name": "Vivid cyan blue",
      "slug": "vivid-cyan-blue",
      "color": "#0693e3"
    },
    {
      "name": "Vivid purple",
      "slug": "vivid-purple",
      "color": "#9b51e0"
    }
  ],
  "theme": [
    {
      "color": "#ffffff",
      "name": "Base",
      "slug": "base"
    },
    {
      "color": "#000000",
      "name": "Contrast",
      "slug": "contrast"
    },
    {
      "color": "#9DFF20",
      "name": "Primary",
      "slug": "primary"
    },
    {
      "color": "#345C00",
      "name": "Secondary",
      "slug": "secondary"
    },
    {
      "color": "#F6F6F6",
      "name": "Tertiary",
      "slug": "tertiary"
    }
  ]
}
```

</details> 

<details><summary>Block Editor View</summary>
<p>

![image](https://user-images.githubusercontent.com/54371619/225860752-6f4eacdb-0b9c-4cb7-9be9-be40561c6d14.png)

</p>
</details> 

Therefore I believe they do not intend for one to supersede the other. The user may select from the `theme` preset or `default` preset at will.

Reply:
Ah, it is not merging associative arrays, so my concern is invalid.